drivers/ukintctlr/xpic/include/uk/intctlr/limits.h

```diff
@@ -6,6 +6,17 @@
 #ifndef __UK_INTCTLR_XPIC_LIMITS_H__
 #define __UK_INTCTLR_XPIC_LIMITS_H__
 
-#define UK_INTCTLR_MAX_IRQ                             16
+#define UK_INTCTLR_FIRST_ALLOCABLE_IRQ			16
+
+#if CONFIG_UKINTCTLR_XPIC_FULL_IDT
+#define UK_INTCTLR_LAST_ALLOCABLE_IRQ			223
+#define UK_INTCTLR_MAX_IRQ				(256 - 32)
+#else /* CONFIG_UKINTCTLR_XPIC_FULL_IDT */
+#define UK_INTCTLR_LAST_ALLOCABLE_IRQ			16
```

Comment:
Probably should have been `15` here.

Reply:
Well then why is this not aligned with what is above?
```C
#define UK_INTCTLR_LAST_ALLOCABLE_IRQ			223 /* 224 - 1 */
#define UK_INTCTLR_MAX_IRQ				224
```
Am I missing something? :thinking:

Reply:
Changing it to `15` would break the subtraction in `IRQ_COUNT`: `LAST - FIRST = 15 - 16 = -1`.

Reply:
Is this not the case for
```C
#define UK_INTCTLR_LAST_ALLOCABLE_IRQ			223 /* 224 - 1 */
#define UK_INTCTLR_MAX_IRQ				224
```
as well?

Reply:
Probably yes.

Reply:
So should `(256 - 32)` be replaced by 223? Where did `(256 - 32)` come from in the first place?

Reply:
Oh hmm, I just noticed that you changed the convention of `MAX_IRQ` to include `MAX_IRQ`.

Reply:
> Where did `(256 - 32)` come from in the first place?

To quote the Intel SDM

> The IA-32 Architecture defines 18 predefined interrupts and exceptions and 224 user defined interrupts, which are
associated with entries in the IDT. Each interrupt and exception in the IDT is identified with a number, called a
vector. Table 6-1 lists the interrupts and exceptions with entries in the IDT and their respective vectors. Vectors 0
through 8, 10 through 14, and 16 through 19 are the predefined interrupts and exceptions; vectors 32 through 255
are for software-defined interrupts, which are for either software interrupts or maskable hardware interrupts

So basically the first 32 are dedicated to implicit exceptions (page faults, alignment check, division by 0, etc) and the rest to 255 can be used for OS defined IRQ's. That is also why the PIC remap has to happen, because by default it generates IRQ's 0 - 15 which overlap with the exceptions, so we make it change this interval to 32 - 47 instead.

Reply:
Okay, so what's the verdict on this?

Reply:
```diff
-#define UK_INTCTLR_LAST_ALLOCABLE_IRQ			223
+#define UK_INTCTLR_LAST_ALLOCABLE_IRQ			224
#define UK_INTCTLR_MAX_IRQ				(256 - 32)
```
Probably this. @mschlumpp

Reply:
That might require some changes in the other code. Because the PR also contains some changes where the interpretation of the interval changed. (I might be wrong though)